### PR TITLE
fix: continue is unnecessary as the last statement in a loop

### PR DESCRIPTION
**Pull request by Hashen110**
Thanks for submitting a pull request! Please provide enough information so that others can review your pull request. The two fields below are mandatory.

If you're making a change to documentation, do **not** modify a `.html` file directly. Instead find the corresponding `.pug` file or test case in the `test/docs` directory.

**Summary**

`continue` is unnecessary as the last statement in a loop

### CODE REVIEW

File: lib/cast.js [hunk 1]
```diff
@@ -317,7 +314,6 @@ module.exports = function cast(schema, obj, options, context) {
                     context: context
                   });
                 }
-                continue;
```
Comment (AbdelrahmanHafez) [inline lib/cast.js:320]: This seems to be in a deeply nested condition, not sure if removing it is okay, frankly, I prefer to just keep it for safety.

I'm curoius, how did you spot those unnecessary `continue` statements? Are you using some kind of tool?

Comment (AbdelrahmanHafez) [inline lib/cast.js:320]: This seems to be in a deeply nested condition, not sure if removing it is okay, frankly, I prefer to just keep it for safety.

I'm curoius, how did you spot those unnecessary `continue` statements? Are you using some kind of tool?

Comment (AbdelrahmanHafez) [reply lib/cast.js:320]: I took a deeper look into this, it should be fine to remove this `continue` as well.
Thanks 👍

**Review by vkarpov15 — COMMENTED** [review]
I'm ok with the `applyProjection()` and `castFilterPath()` changes, but not the other two. The changes for `cast.js` and `document.js` are a bit risky IMO because those `for` loops have very long and cumbersome `if` statements, which is our fault. Even if it is ok to remove `continue` there, we may add code in the future that will change that, so I'd rather err on the side of caution unless there's some reason (maybe performance) to prefer removing `continue`?

File: lib/cast.js (hunk 1 above)
Comment (Hashen110) [reply lib/cast.js:320]: IDE (WebStorm) suggested as a warning.

**Comment by vkarpov15** [comment]
I'm going to close this PR for now. Thanks for the contribution, but I don't see enough benefit to risk merging this. Granted that's mostly our fault, we should refactor our code to avoid having such long `for` loops.

**Pull request closed without merging**